c/zss.c

```diff
@@ -805,6 +842,8 @@ static WebPluginListElt* readWebPluginDefinitions(HttpServer *server, ShortLived
                           htPut(server->loggingIdsByName,
                                 key,
                                 &(plugin->dataServices[i]->loggingIdentifier));
+                          checkAndSetServiceLoglevel(serverConfigFile, plugin->dataServices[i]->identifier, 
```

Comment:
You could pass `key` instead of `plugin->dataServices[i]->identifier` to avoid the code duplication inside of `checkAndSetServiceLoglevel`.

Reply:
there is different formatting for key but i can reuse existing logic. Reused key formation logic with additional buffer.

Reply:
key has plugin name and service name concatenated by `:` but in config loglevels it is expected to be concatenated by `.`
so key can not be used as it is in checkAndSetServiceLoglevel().

Reply:
right, so you could make a new function that accepts `:` or `.` as a parameter.

Reply:
right. but anyway you could create a new function that accepts the char (`:` or `.`)  as a parameter.

Reply:
ok, good to know that we have flexibility to refactor legacy code.